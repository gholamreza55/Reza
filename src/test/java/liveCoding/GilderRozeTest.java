package liveCoding;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GilderRozeTest {
    @Test
    void should_update_quality() {
        String[] name = new String[]{"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert"};
        Integer[] selin = new Integer[]{0, -1, 1, 11, 12};
        Integer[] quality = new Integer[]{0, 1, 2, 50, 49, -1};
        //Item output = Updatability(foo, selin, quality);
        CombinationApprovals.verifyAllCombinations(this::Updatequlity, name, selin, quality);
    }

    private String Updatequlity(String name, int selin, int quality) {
        Item[] items = new Item[]{new Item(name, selin, quality)};
        GilderRoze app = new GilderRoze(items);
        app.updatequality();
       // Item output = app.items[0];
        return app.items[0].toString();
    }
}
