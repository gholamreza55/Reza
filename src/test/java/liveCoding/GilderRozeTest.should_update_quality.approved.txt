[foo, 0, 0] => Item{name='foo', selin=0, quality=0} 
[foo, 0, 1] => Item{name='foo', selin=0, quality=0} 
[foo, 0, 2] => Item{name='foo', selin=0, quality=1} 
[foo, 0, 50] => Item{name='foo', selin=0, quality=49} 
[foo, 0, 49] => Item{name='foo', selin=0, quality=48} 
[foo, 0, -1] => Item{name='foo', selin=0, quality=-1} 
[foo, -1, 0] => Item{name='foo', selin=-1, quality=0} 
[foo, -1, 1] => Item{name='foo', selin=-1, quality=0} 
[foo, -1, 2] => Item{name='foo', selin=-1, quality=1} 
[foo, -1, 50] => Item{name='foo', selin=-1, quality=49} 
[foo, -1, 49] => Item{name='foo', selin=-1, quality=48} 
[foo, -1, -1] => Item{name='foo', selin=-1, quality=-1} 
[foo, 1, 0] => Item{name='foo', selin=1, quality=0} 
[foo, 1, 1] => Item{name='foo', selin=1, quality=0} 
[foo, 1, 2] => Item{name='foo', selin=1, quality=1} 
[foo, 1, 50] => Item{name='foo', selin=1, quality=49} 
[foo, 1, 49] => Item{name='foo', selin=1, quality=48} 
[foo, 1, -1] => Item{name='foo', selin=1, quality=-1} 
[foo, 11, 0] => Item{name='foo', selin=11, quality=0} 
[foo, 11, 1] => Item{name='foo', selin=11, quality=0} 
[foo, 11, 2] => Item{name='foo', selin=11, quality=1} 
[foo, 11, 50] => Item{name='foo', selin=11, quality=49} 
[foo, 11, 49] => Item{name='foo', selin=11, quality=48} 
[foo, 11, -1] => Item{name='foo', selin=11, quality=-1} 
[foo, 12, 0] => Item{name='foo', selin=12, quality=0} 
[foo, 12, 1] => Item{name='foo', selin=12, quality=0} 
[foo, 12, 2] => Item{name='foo', selin=12, quality=1} 
[foo, 12, 50] => Item{name='foo', selin=12, quality=49} 
[foo, 12, 49] => Item{name='foo', selin=12, quality=48} 
[foo, 12, -1] => Item{name='foo', selin=12, quality=-1} 
[Aged Brie, 0, 0] => Item{name='Aged Brie', selin=0, quality=1} 
[Aged Brie, 0, 1] => Item{name='Aged Brie', selin=0, quality=2} 
[Aged Brie, 0, 2] => Item{name='Aged Brie', selin=0, quality=3} 
[Aged Brie, 0, 50] => Item{name='Aged Brie', selin=0, quality=50} 
[Aged Brie, 0, 49] => Item{name='Aged Brie', selin=0, quality=50} 
[Aged Brie, 0, -1] => Item{name='Aged Brie', selin=0, quality=0} 
[Aged Brie, -1, 0] => Item{name='Aged Brie', selin=-1, quality=0} 
[Aged Brie, -1, 1] => Item{name='Aged Brie', selin=-1, quality=0} 
[Aged Brie, -1, 2] => Item{name='Aged Brie', selin=-1, quality=0} 
[Aged Brie, -1, 50] => Item{name='Aged Brie', selin=-1, quality=0} 
[Aged Brie, -1, 49] => Item{name='Aged Brie', selin=-1, quality=0} 
[Aged Brie, -1, -1] => Item{name='Aged Brie', selin=-1, quality=0} 
[Aged Brie, 1, 0] => Item{name='Aged Brie', selin=1, quality=1} 
[Aged Brie, 1, 1] => Item{name='Aged Brie', selin=1, quality=2} 
[Aged Brie, 1, 2] => Item{name='Aged Brie', selin=1, quality=3} 
[Aged Brie, 1, 50] => Item{name='Aged Brie', selin=1, quality=50} 
[Aged Brie, 1, 49] => Item{name='Aged Brie', selin=1, quality=50} 
[Aged Brie, 1, -1] => Item{name='Aged Brie', selin=1, quality=0} 
[Aged Brie, 11, 0] => Item{name='Aged Brie', selin=11, quality=1} 
[Aged Brie, 11, 1] => Item{name='Aged Brie', selin=11, quality=2} 
[Aged Brie, 11, 2] => Item{name='Aged Brie', selin=11, quality=3} 
[Aged Brie, 11, 50] => Item{name='Aged Brie', selin=11, quality=50} 
[Aged Brie, 11, 49] => Item{name='Aged Brie', selin=11, quality=50} 
[Aged Brie, 11, -1] => Item{name='Aged Brie', selin=11, quality=0} 
[Aged Brie, 12, 0] => Item{name='Aged Brie', selin=12, quality=1} 
[Aged Brie, 12, 1] => Item{name='Aged Brie', selin=12, quality=2} 
[Aged Brie, 12, 2] => Item{name='Aged Brie', selin=12, quality=3} 
[Aged Brie, 12, 50] => Item{name='Aged Brie', selin=12, quality=50} 
[Aged Brie, 12, 49] => Item{name='Aged Brie', selin=12, quality=50} 
[Aged Brie, 12, -1] => Item{name='Aged Brie', selin=12, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=0, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=0, quality=4} 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=0, quality=5} 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=0, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=0, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=0, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=-1, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=-1, quality=4} 
[Backstage passes to a TAFKAL80ETC concert, -1, 2] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=-1, quality=5} 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=-1, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=-1, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=-1, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=1, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=1, quality=4} 
[Backstage passes to a TAFKAL80ETC concert, 1, 2] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=1, quality=5} 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=1, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=1, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 1, -1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=1, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=11, quality=1} 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=11, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=11, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=11, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=11, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 11, -1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=11, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=12, quality=1} 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=12, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 12, 2] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=12, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=12, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=12, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 12, -1] => Item{name='Backstage passes to a TAFKAL80ETC concert', selin=12, quality=0} 
